SpiralCompiler/Syntax/Lexer.cs

```diff
@@ -0,0 +1,217 @@
+
+            var ch = Peek();
+            if (char.IsWhiteSpace(ch))
+            {
+                ++index;
+                goto begin;
+            }
+
+            // Comment
+            if (ch == '/' && Peek(1) == '/')
+            {
+                index += 2;
+                while (Peek(defaultChar: '\r') != '\r' && Peek(defaultChar: '\n') != '\n')
+                {
+                    ++index;
+                }
+                goto begin;
+            }
+
+            // string
+            if (ch == '"')
+            {
+                var offset = 1;
+                while (Peek(offset, defaultChar: '"') != '"') ++offset;
+                ++offset;
+                return Consume(offset, TokenType.String);
+            }
+
+            // braces
+            if (ch == '{') return Consume(1, TokenType.OpenBrace);
+            if (ch == '}') return Consume(1, TokenType.CloseBrace);
+
+            // parens
+            if (ch == '(') return Consume(1, TokenType.OpenParen);
+            if (ch == ')') return Consume(1, TokenType.CloseParen);
+
+            // comma
+            if (ch == ',') return Consume(1, TokenType.Comma);
+
+            // semicolon
+            if (ch == ';') return Consume(1, TokenType.Semicolon);
+
+            // <
+            if (ch == '<')
+            {
+                // <=
+                if (Peek(1) == '=')
+                {
+                    return Consume(2, TokenType.LessEquals);
+                }
+                return Consume(1, TokenType.LessThan);
+            }
+
+            // >
+            if (ch == '>')
+            {
+                // >=
+                if (Peek(1) == '=')
+                {
+                    return Consume(2, TokenType.GreaterEquals);
+                }
+                return Consume(1, TokenType.GreaterThan);
+            }
+
+            // == and =
+            if (ch == '=')
+            {
+                if (Peek(1) == '=') return Consume(2, TokenType.Equals);
+                else return Consume(1, TokenType.Assign);
+            }
+
+            // +
+            if (ch == '+')
+            {
+                // ++
+                if (Peek(1) == '+')
+                {
+                    return Consume(2, TokenType.Increment);
+                }
+                return Consume(1, TokenType.Plus);
+            }
+
+            // -
+            if (ch == '-')
+            {
+                if (Peek(1) == '-')
+                {
+                    return Consume(2, TokenType.Decrement);
+                }
+                return Consume(1, TokenType.Minus);
+            }
+
+            // *
+            if (ch == '*') return Consume(1, TokenType.Multiply);
+
+            // /
+            if (ch == '/') return Consume(1, TokenType.Divide);
+
+            // range
+            if (ch == '.' && Peek(1) == '.')
+            {
+                return Consume(2, TokenType.Range);
+            }
```

Comment:
Why is dot so far apart from range?
For all the other cases you grouped based on the first character.